activation/handler_v2.go

```diff
@@ -519,6 +605,9 @@ func (h *HandlerV2) syntacticallyValidateDeps(
 	}
 
 	if atx.Initial == nil {
+		if smesherCommitment == nil {
+			return nil, nil, errors.New("ATX signer not present in merged ATX")
+		}
 		err := h.nipostValidator.VRFNonceV2(atx.SmesherID, *smesherCommitment, atx.VRFNonce, atx.TotalNumUnits())
```

Comment:
For the weight we have to be a bit more nuanced - either we consider the weight the total number of units in the ATX * the number of ticks of the lowest height poet proof or (better) calculate the weight for each identity individually and sum it up.

Reply:
Updated weight calculation